lib/src/Uncertainty/Distribution/UniformOrderStatistics.cxx

```diff
@@ -0,0 +1,470 @@
+}
+
+/* Compute the mean of the distribution */
+void UniformOrderStatistics::computeMean() const
+{
+  mean_ = Point(dimension_);
+  for (UnsignedInteger i = 0; i < dimension_; ++i)
+    mean_[i] =  (i + 1.0) / (dimension_ + 1.0);
+  isAlreadyComputedMean_ = true;
+}
+
+/* Compute the entropy of the distribution */
+Scalar UniformOrderStatistics::computeEntropy() const
+{
+  return -SpecFunc::LogGamma(dimension_ + 1.0);
+}
+
+/* Get the standard deviation of the distribution */
+Point UniformOrderStatistics::getStandardDeviation() const
+{
+  Point sigma(dimension_);
+  for (UnsignedInteger i = 0; i < dimension_; ++i)
+    sigma[i] = 1.0 / (dimension_ + 1.0) * std::sqrt((i + 1.0) * (dimension_ - i) / (dimension_ + 2.0));
+  return sigma;
+}
+
+/* Get the skewness of the distribution */
+Point UniformOrderStatistics::getSkewness() const
+{
+  Point skewness(dimension_);
+  for (UnsignedInteger i = 0; i < dimension_; ++i)
+    skewness[i] = 2.0 * (dimension_ - 1.0 - 2.0 * i) / (dimension_ + 3.0) * std::sqrt((dimension_ + 2.0) / ((i + 1.0) * (dimension_ - i)));
+  return skewness;
+}
+
+/* Get the kurtosis of the distribution */
+Point UniformOrderStatistics::getKurtosis() const
+{
+  Point kurtosis(dimension_);
+  for (UnsignedInteger i = 0; i < dimension_; ++i)
+    kurtosis[i] = 3.0 * (dimension_ + 2.0) * (2.0 * (dimension_ + 1.0) * (dimension_ + 1.0) + (i + 1.0) * ((dimension_ + 1.0) - 6.0) * (dimension_ - i)) / ((i + 1.0) * (dimension_ - i) * (3.0 + (dimension_ + 1.0)) * (2.0 + (dimension_ + 1.0)));
+  return kurtosis;
+}
+
+/* Compute the covariance of the distribution */
+void UniformOrderStatistics::computeCovariance() const
+{
+  DistributionImplementation::computeCovarianceGeneral();
+  isAlreadyComputedCovariance_ = true;
+}
+
+/* Get the quantile of the distribution */
+Scalar UniformOrderStatistics::computeScalarQuantile(const Scalar prob,
+    const Bool tail) const
+{
+  if (tail) return 1.0 - prob;
+  return prob;
+}
+
+/* Get the i-th marginal distribution */
+Distribution UniformOrderStatistics::getMarginal(const UnsignedInteger i) const
+{
+  if (i >= dimension_) throw InvalidArgumentException(HERE) << "The index of a marginal distribution must be in the range [0, dim-1]";
+  return getMarginal(Indices(1, i));
+}
+
+/* Get the distribution of the marginal distribution corresponding to indices dimensions */
+Distribution UniformOrderStatistics::getMarginal(const Indices & indices) const
+{
+  const UnsignedInteger dimension = getDimension();
+  if (!indices.check(dimension)) throw InvalidArgumentException(HERE) << "The indices of a marginal distribution must be in the range [0, dim-1] and must be different";
+  if (dimension == 1) return clone();
+  if (indices.getSize() == 1)
+  {
+    const UnsignedInteger i = indices[0];
+    Beta marginal(i + 1.0, dimension_ - i, 0.0, 1.0);
+    marginal.setDescription(Description(1, getDescription()[i]));
+    return marginal;
+  }
+  return DistributionImplementation::getMarginal(indices);
```

Comment:
According to (Arnold et al., 2008) pages 19 (eq. 2.3.12) and page 78, in the case where d = 2, then the distribution is Bivariate Beta(i, j - i, n - j = 1). Is this distribution in the catalogue?

Reply:
We don't have this in stock! The  closest is the bivariate Dirichlet distribution, but there is a transformation to be applied to the arguments of the PDF to have the correct support.
BTW I am waiting for https://github.com/openturns/openturns/pull/2603 to be merged in order to implement a dedicated distribution for these multivariate marginal distributions. We already detailed the formulas for the PDF and CDF with @adutfoy and we need the new version of MarginalDistribution to have a base for these new distributions.